src/main/java/sparksoniq/spark/iterator/flowr/LetClauseSparkIterator.java

```diff
@@ -72,10 +162,29 @@ public LetClauseSparkIterator(RuntimeTupleIterator child, VariableReferenceItera
             expression.close();
             List<FlworTuple> tuples = new ArrayList<>();
             FlworTuple tuple  = new FlworTuple();
-            tuple.putValue(((VariableReferenceIterator)this._children.get(0)).getVariableName(), contents, false);
+            tuple.putValue(_variableReference.getVariableName(), contents, false);
             tuples.add(tuple);
             rdd = SparkContextManager.getInstance().getContext().parallelize(tuples);
         }
         return rdd;
     }
+
+
+    @Override
+    public boolean hasNext() {
+        return super.hasNext();
+    }
+
+    @Override
+    public void reset(DynamicContext context) {
+        super.reset(context);
```

Comment:
This is superflous (the override only calls the super function which only adds a performance overhead)